src/discussions/comments/data/api.js

```diff
@@ -75,14 +75,27 @@ export async function postComment(comment, threadId, parentId) {
 /**
  * Updates existing comment.
  * @param {string} commentId ID of comment to update.
- * @param {string} comment Raw updated comment data to post.
+ * @param {string=} comment Raw updated comment data to post.
+ * @param {boolean=} voted
+ * @param {boolean=} flagged
+ * @param {boolean=} endorsed
  * @returns {Promise<{}>}
  */
-export async function updateComment(commentId, comment) {
+export async function updateComment(commentId, {
+  comment,
+  voted,
+  flagged,
+  endorsed,
+}) {
   const url = `${commentsApiUrl}${commentId}/`;
 
   const { data } = await getAuthenticatedHttpClient()
-    .patch(url, { raw_body: comment }, { headers: { 'Content-Type': 'application/merge-patch+json' } });
+    .patch(url, {
+      raw_body: comment,
+      voted,
+      abuse_flagged: flagged,
+      endorsed,
```

Comment:
@xitij2000 might be a little trivial but I think separating the data here from the API call makes it a little bit more readable